### PR TITLE
Refactor ex_getln.c to remove calls to STRLEN()

### PR DESCRIPTION
**Pull request by basilisk0315**
Also add dealloc_cmdbuf() to keep ccline.cmdlen and ccline.cmdbuflen in sync with ccline.cmdbuff.

### CODE REVIEW

**Comment by chrisbra** [comment]
thanks!

File: src/ex_getln.c [hunk 1]
```diff
@@ -4684,21 +4729,30 @@ open_cmdwin(void)
...
-		ccline.cmdbuff = vim_strsave((char_u *)p);
+		ccline.cmdbuff = vim_strnsave(p, plen);
+		ccline.cmdlen = plen;
+		ccline.cmdbufflen = plen + 1;
 		cmdwin_result = CAR;
 	    }
 	    else
 	    {
 		// First need to cancel what we were doing.
-		ccline.cmdbuff = NULL;
```
Comment (zeertzjq) [inline src/ex_getln.c:4701]: Why is this line removed?

Comment (zeertzjq) [reply src/ex_getln.c:4701]: Oh I see. `dealloc_cmdbuff()` was already called above.
